src/reanimated2/hook/useScrollViewOffset.ts

```diff
@@ -18,25 +18,26 @@ const scrollEventNames = [
 ];
 
 export function useScrollViewOffset(
-  aref: RefObject<Animated.ScrollView>,
+  aref: RefObject<AnimatedScrollView>,
   initialRef?: SharedValue<number>
 ): SharedValue<number> {
   const offsetRef = useRef(
     initialRef !== undefined ? initialRef : useSharedValue(0)
   );
 
-  const event = useEvent<ScrollEvent>((event: ScrollEvent) => {
-    'worklet';
-    offsetRef.current.value =
-      event.contentOffset.x === 0
-        ? event.contentOffset.y
-        : event.contentOffset.x;
-  }, scrollEventNames);
+  const event = useEvent<RNNativeScrollEvent>(
+    (event: ReanimatedScrollEvent) => {
+      'worklet';
+      offsetRef.current.value =
+        event.contentOffset.x === 0
+          ? event.contentOffset.y
+          : event.contentOffset.x;
+    },
+    scrollEventNames
+  ) as unknown as EventHandlerInternal<ReanimatedScrollEvent>;
```

Comment:
How does it work? EventHandlerInternal is of type
```
current: something
```
and `useEvent` return `EventHandlerProcessed` of type `(event: Event, context?: Context) => void;`

Reply:
The logic is handled inside `createAnimatedComponent`. If you look there you'll find:
![Screenshot 2023-09-18 at 16 01 18](https://github.com/software-mansion/react-native-reanimated/assets/40713406/03c3ad3d-7f4d-4b63-af70-35e5921f35e8)